lib/webrick/httpresponse.rb

```diff
@@ -517,14 +539,16 @@ def send_body_proc(socket)
         @body.call(ChunkedWrapper.new(socket, self))
         socket.write("0#{CRLF}#{CRLF}")
       else
-        size = @header['content-length'].to_i
         if @bodytempfile
           @bodytempfile.rewind
           IO.copy_stream(@bodytempfile, socket)
         else
           @body.call(socket)
         end
-        @sent_size = size
+
+        if content_length = @header['content-length']
```

Comment:
Turns out that since `@sent_size` is set to `0` in initialize, this doesn't currently have an effect.  Still, this conditional seems like a better approach.

Reply:
Yes I'm fine with that behaviour, or alternatively we could set it to some other value, e.g. `nil` to represent "unknown".